src/transformers/pipelines/text_classification.py

```diff
@@ -64,11 +92,38 @@ def __call__(self, *args, **kwargs):
         """
         outputs = super().__call__(*args, **kwargs)
 
-        if self.model.config.num_labels == 1:
-            scores = 1.0 / (1.0 + np.exp(-outputs))
+        return_all_scores = return_all_scores if return_all_scores is not None else self.return_all_scores
+        function_to_apply = function_to_apply if function_to_apply is not None else self.function_to_apply
+
+        if function_to_apply is None and self.model.config.problem_type is not None:
+            if self.model.config.problem_type == "multi_label_classification":
+                self.function_to_apply = "sigmoid"
+            elif self.model.config.problem_type == "single_label_classification":
+                self.function_to_apply = "softmax"
+
+        def sigmoid(_outputs):
+            return 1.0 / (1.0 + np.exp(-_outputs))
+
+        def softmax(_outputs):
+            maxes = np.max(_outputs, axis=-1, keepdims=True)
+            shifted_exp = np.exp(_outputs - maxes)
+            return shifted_exp / shifted_exp.sum(axis=-1, keepdims=True)
+
+        if function_to_apply == "default":
```

Comment:
I think we should stay away from "default" in that case.

IMO this should be taken care of when inferring the `function_to_apply` this will make logic a bit more simple to understand.